### PR TITLE
Pytest conversion for api/lifecycleenvironment

### PR DESCRIPTION
**Pull request by latran**
Convert all test cases into pytest for api/lce

### CODE REVIEW

File: tests/foreman/api/test_lifecycleenvironment.py [hunk 1]
```diff
@@ -24,204 +24,203 @@
...
 
-        :CaseImportance: Critical
-        """
-        lc_env = entities.LifecycleEnvironment(organization=self.org).create()
-        lc_env.delete()
-        with self.assertRaises(HTTPError):
-            lc_env.read()
+    :CaseImportance: Low
+    """
+    module_lce = entities.LifecycleEnvironment(description=gen_string('alpha')).create()
```
Comment (jameerpathan111) [inline tests/foreman/api/test_lifecycleenvironment.py:137]: @latran is it required to create new lce? In `test_positive_update_name` test you didn't created one.

File: tests/foreman/api/test_lifecycleenvironment.py [hunk 2]
```diff
@@ -24,204 +24,203 @@
...
+    :id: 66d34781-8210-4282-8b5e-4be811d5c756
 
-        :CaseImportance: Low
-        """
-        for name in invalid_values_list():
-            with self.subTest(name):
-                with self.assertRaises(HTTPError):
-                    entities.LifecycleEnvironment(name=name).create()
+    :expectedresults: Lifecycle environment is created with Library as
+        prior
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_lifecycleenvironment.py:83]: Aren't these descriptions inconsistent?

If something is "prior to Library", then it's first "something" and then Library.

If something is created with "Library as prior", then it's first Library, and then "something".

Or is there some language nuance that I am missing?

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 1 above)
Comment (mirekdlugosz) [reply tests/foreman/api/test_lifecycleenvironment.py:137]: Agreed with @jameerpathan111 , you already have `lce` fixture that creates new lce for each test (that uses it).

File: tests/foreman/api/test_lifecycleenvironment.py [hunk 3]
```diff
@@ -24,204 +24,203 @@
...
+
+    :expectedresults: Only "Library" and the lifecycle environment just
+        created are in the search results.
+
+    :CaseLevel: Integration
+    """
+    lc_env = entities.LifecycleEnvironment(organization=module_org).create()
+    lc_envs = lc_env.search({'organization'})
+    assert len(lc_envs) == 2
+    assert {lc_env_.name for lc_env_ in lc_envs}, {u'Library', lc_env.name}
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_lifecycleenvironment.py:201]: Does that work?

module_org is module scoped, so all your tests here will share single organization. Unless this is very first test executed, I would expect this organization to have multiple lifecycle environments.

I'm afraid you need to create organization explicitly for this test, either inside function or using function-scoped fixture.

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 2 above)
Comment (latran) [reply tests/foreman/api/test_lifecycleenvironment.py:83]: It's the latter.  "Library as prior" as you've stated so it's first Library then "something"

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 1 above)
Comment (latran) [reply tests/foreman/api/test_lifecycleenvironment.py:137]: The `lce` fixture has no description.  I created `module_lce` to create a description and then update it.

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 3 above)
Comment (latran) [reply tests/foreman/api/test_lifecycleenvironment.py:201]: This does work, but seems to fail for the stand alone jenkins run.  I'll see what if I can fix this.

File: tests/foreman/api/test_lifecycleenvironment.py [hunk 4]
```diff
@@ -24,204 +24,203 @@
...
+
+
+@tier2
+@pytest.mark.stubbed('Implement once BZ1348727 is fixed')
+def test_positive_create_environment_after_host_register(self):
```
Comment (jameerpathan111) [inline tests/foreman/api/test_lifecycleenvironment.py:206]:
```suggestion
def test_positive_create_environment_after_host_register():
```

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 2 above)
Comment (mirekdlugosz) [reply tests/foreman/api/test_lifecycleenvironment.py:83]: Could you please change first line in dosctring, so it's consistent with expectedresults and your comment?

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 1 above)
Comment (mirekdlugosz) [reply tests/foreman/api/test_lifecycleenvironment.py:137]: @latran I'm leaving that up to you, but from my point of view, it makes sense for `lce` fixture to create lce with all params, so update tests can focus on actual updates.

**Comment by latran** [comment]
Test result of file below.
```
pytest tests/foreman/api/test_lifecycleenvironment.py 
=========================================================================================================================== test session starts ============================================================================================================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ltran/Projects/robottelo
plugins: services-1.3.1, forked-1.3.0, mock-1.10.4, xdist-2.1.0, cov-2.10.1
collecting ... 2020-10-05 15:25:06 - conftest - DEBUG - Collected 10 test cases
collected 10 items                                                                                                                                                                                                                                                         

tests/foreman/api/test_lifecycleenvironment.py .........s                                                                                                                                                                                                            [100%]

============================================================================================================================= warnings summary =============================================================================================================================
/home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337
  /home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.lifecycleenvironments - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

/home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337
  /home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.critical - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

/home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337
  /home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.low - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

/home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337
  /home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.high - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

/home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337
  /home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.This is a Manual test! - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

/home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337
  /home/ltran/Projects/venv/robo_venv/lib/python3.7/site-packages/_pytest/mark/structures.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.BZ_1348727 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================================================================================= 9 passed, 1 skipped, 6 warnings in 35.80 seconds =======================
```

File: tests/foreman/api/test_lifecycleenvironment.py (hunk 1 above)
Comment (latran) [reply tests/foreman/api/test_lifecycleenvironment.py]: Added description to fixture.

**Review by mirekdlugosz — APPROVED** [review]
My comments are resolved, ACK.

**Comment by latran** [comment]
@jameerpathan111 need your review.

**Review by jameerpathan111 — APPROVED** [review]
ACK

**Pull request merged**